src/Snake.tsx

```diff
@@ -5,23 +5,67 @@ type Point = {
   x: number;
   y: number;
 };
+type Props = {
+  pokemonId: Pokemon['id'];
+};
 
-const GRID_SIZE = 21;
+const GRID_ROWS_SIZE = 21;
+const GRID_COLUMNS_SIZE = 25;
 
-export const SnakeStates = ({ pokemonId }: { pokemonId: Pokemon['id'] }): JSX.Element => {
+export const SnakeStates = ({ pokemonId }: Props): JSX.Element => {
+  const [speed, setSpeed] = useState(6);
   const [snakeBody, setSnakeBody] = useState<Point[]>([
-    { x: 11, y: 4 },
-    { x: 11, y: 3 },
+    { x: 4, y: 11 },
+    { x: 3, y: 11 },
   ]);
   const [food, setFood] = useState<Point>(generatePointOutSnakeBody(snakeBody));
-
+  const [direction, setDirection] = useState<Point>({ x: 1, y: 0 });
   const getElementPositionStyle = (element: Point) => {
     return {
-      gridRowStart: element.x,
-      gridColumnStart: element.y,
+      gridRowStart: element.y,
+      gridColumnStart: element.x,
     };
   };
 
+  useEffect(() => {
+    const keyPressHandler = (ev: KeyboardEvent) => {
+      switch (ev.code) {
+        case 'ArrowUp':
+          setDirection((prev) => {
+            return prev.y === 1 ? prev : { x: 0, y: -1 };
+          });
```

Comment:
This conditio is better because it doesn't mutate state from `{ x: 0, y: -1 }` to a different but same by mining object `{ x: 0, y: -1 }` when `y` equals `-1`. Please do similar change for all the arrows
```suggestion
          setDirection(({ y }) => {
            return y === 0 ? { x: 0, y: -1 } : prev;
          });
```